nilearn/plotting/img_plotting.py

```diff
@@ -37,6 +39,43 @@
 # Core, usage-agnostic functions
 
 
+def check_threshold(thr, data):
+    """
+    Checks that the given threshold has an accepted string value
+
+    Parameters
+    ----------
+    thr: str or a list of strings or number or list of numbers
+        If threshold is a string itmush finish with a percent sign,
+        e.g. "99.7%", or if it a value it can be a real numbers.
+    data: a Numpy array of a map to calculate the
+        threshold using a string percentage value.
+
+    Returns
+    -------
+    returns threshold as an output also checks if the value entered
+    is valid
+
+    """
+    if isinstance(thr, _basestring):
+        message = ("If 'threshold' is given as string it "
+                   'should be a number followed by the percent'
+                   'sign, "e.g. 25.3%"')
+        if not thr.endswith('%'):
+            raise ValueError(message)
+        try:
+            percentile = float(thr[:-1])
+            thr = fast_abs_percentile(data, percentile) + 1e-6
```

Comment:
I see the problem you are addressing but I don't think it's the best way to do it.

Your problem is : when the percentile il too high, all values of the image are kept, including the 0. in the background.
For me, the solution would be to consider the background to be masked by default, because it's background. And this is not specific to the percentile mode. I would remove the `1e-6` and add, just before the return statement:
```python
# Get rid of background values in all cases
thr = max(thr, 1e-6)
```